Modules/@babylonjs/react-native-iosandroid/android/src/main/java/com/babylonreactnative/EngineView.java

```diff
@@ -67,34 +71,50 @@ public void surfaceDestroyed(SurfaceHolder holder) {
     public void setAntiAliasing(Integer value) {
         BabylonNativeInterop.updateMSAA(value);
     }
+
+    public void setIsTopMost(Boolean isTopMost) {
+        setIsTransparentAndIsTopMost(this.isTransparent, isTopMost);
+    }
     // ------------------------------------
     // TextureView related
-
     public void setIsTransparent(Boolean isTransparent) {
-        if (isTransparent) {
-            if (this.opaqueSurfaceView != null) {
-                this.opaqueSurfaceView.setVisibility(View.GONE);
-                this.opaqueSurfaceView = null;
-            }
-            if (this.transparentTextureView == null) {
-                this.transparentTextureView = new TextureView(this.getContext());
-                this.transparentTextureView.setLayoutParams(EngineView.childViewLayoutParams);
-                this.transparentTextureView.setSurfaceTextureListener(this);
-                this.transparentTextureView.setOpaque(false);
-                this.addView(this.transparentTextureView);
-            }
+        setIsTransparentAndIsTopMost(isTransparent, this.isTopMost);
+    }
+
+    private void setIsTransparentAndIsTopMost(Boolean isTransparent, Boolean isTopMost) {
```

Comment:
Should we be checking the inputs against the current values and not recreating views when unnecessary?

Reply:
Fixed.